src/azure/azure-mesh-adapter.ts

```diff
@@ -110,16 +114,89 @@ export class AzureMeshAdapter implements MeshAdapter {
           platform: MeshPlatform.Azure,
           nativeObj: sub,
           costs: [],
+          roleAssignments: [],
         };
       }),
     );
   }
 
+  async loadTenantRoleAssignments(tenants: MeshTenant[]): Promise<void> {
+    // Only work on Azure tenants
+    const azureTenants = tenants.filter((t) => isSubscription(t.nativeObj));
+
+    for (const t of azureTenants) {
+      const roleAssignments = await this.loadRoleAssignmentsForTenant(t);
+      t.roleAssignments.push(...roleAssignments);
+    }
+  }
+
+  private async loadRoleAssignmentsForTenant(
+    tenant: MeshTenant,
+  ): Promise<MeshTenantRoleAssignment[]> {
+    if (!isSubscription(tenant.nativeObj)) {
+      throw new MeshError(
+        "Given tenant did not contain an Azure Subscription native object",
+      );
+    }
+
+    const roleAssignments = await this.azureCli.getRoleAssignments(
+      tenant.nativeObj,
+    );
+
+    return roleAssignments.map((x) => {
+      const { assignmentSource, assignmentId } = this.getAssignmentFromScope(x.scope);
+      return {
+        principalId: x.principalId,
+        principalName: x.principalName,
+        principalType: this.toMeshPrincipalType(x.principalType),
+        roleId: x.roleDefinitionId,
+        roleName: x.roleDefinitionName,
+        assignmentSource,
+        assignmentId
+      };
+    });
+  }
+
+  private toMeshPrincipalType(principalType: string): MeshPrincipalType {
+    switch (principalType) {
+      case "User":
+        return MeshPrincipalType.User;
+      case "Group":
+        return MeshPrincipalType.Group;
+      case "ServicePrincipal":
+        return MeshPrincipalType.TechnicalUser;
+      default:
+        throw new MeshError(
+          "Found unknown principalType for Azure: " + principalType,
+        );
+    }
+  }
+
   private convertTags(tags: Tag[]): MeshTag[] {
     return tags.map<MeshTag>((t) => {
       const tagValues = t.values.map((tv) => tv.tagValue);
 
       return { tagName: t.tagName, tagValues };
     });
   }
+
+  private getAssignmentFromScope(scope: string): { assignmentId: string, assignmentSource: MeshRoleAssignmentSource } {
+    const map: { [key in MeshRoleAssignmentSource]: RegExp } = {
+      Organization: /^\/$/, // This means string should equal exactly to "/".
+      Ancestor: /\/managementGroups\//,
+      Tenant: /\/subscriptions\//
+    };
+    for (let key in map) {
+      if (map.hasOwnProperty(key)) {
```

Comment:
d: is this check not superficial? We iterate over all keys in map, how can there be a property not be in map? As this object is not inherited or anything but only a simple object.

Reply:
t: three if ... else statements in a row also sometimes do the trick. :P ;)

Reply:
This is such a stupid check that still exists in JS in 2021... I've always hated it and find it super unnecessary. I will remove it and just ignore IntelliJ complaining about it